src/Symfony/Component/Notifier/Bridge/Engagespot/EngagespotTransportFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Engagespot;
+
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Daniel GORGAN <https://github.com/danut007ro>
+ */
+final class EngagespotTransportFactory extends AbstractTransportFactory
+{
+    /**
+     * @return EngagespotTransport
+     */
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if ('engagespot' !== $scheme) {
+            throw new UnsupportedSchemeException($dsn, 'engagespot', $this->getSupportedSchemes());
+        }
+
+        $apiKey = $dsn->getUser();
+        $campaignName = $dsn->getOption('campaign_name');
```

Comment:
```suggestion
        $campaignName = $dsn->getRequiredOption('campaign_name');
```
Otherwise it can be null, if it is optional please change the construct method in the transport and make the toString method only print the camapign_name if set.

Thanks